dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -1018,6 +1018,57 @@ TRASH_TrashFile(LPCWSTR wszPath)
     return DeleteFileToRecycleBin(wszPath);
 }
 
+static void TRASH_PlayEmptyRecycleBinSound()
+{
+    CRegKey regKey;
+    CHeapPtr<WCHAR> pszValue;
+    CHeapPtr<WCHAR> pszSndPath;
+    DWORD dwType, dwSize;
+    LONG lError;
+
+    lError = regKey.Open(HKEY_CURRENT_USER,
+                         L"AppEvents\\Schemes\\Apps\\Explorer\\EmptyRecycleBin\\.Current",
+                         KEY_READ);
+    if (lError != ERROR_SUCCESS)
+        return;
+
+    lError = regKey.QueryValue(NULL, &dwType, NULL, &dwSize);
+    if (lError != ERROR_SUCCESS)
+        return;
+
+    if (!pszValue.AllocateBytes(dwSize))
+        return;
+
+    lError = regKey.QueryValue(NULL, &dwType, pszValue, &dwSize);
+    if (lError != ERROR_SUCCESS)
+        return;
+
+    if (dwType == REG_EXPAND_SZ)
+    {
+        dwSize = ExpandEnvironmentStringsW(pszValue, NULL, 0);
+        if (dwSize == 0)
+            return;
+
+        if (!pszSndPath.Allocate(dwSize))
+            return;
+
+        if (ExpandEnvironmentStringsW(pszValue, pszSndPath, dwSize) == 0)
+            return;
+    }
+    else if (dwType == REG_SZ)
+    {
+        /* The type is REG_SZ, no need to expand */
+        pszSndPath.Attach(pszValue.Detach());
+    }
+    else
+    {
+        /* Invalid type */
+        return;
+    }
+
+    PlaySoundW(pszSndPath, NULL, SND_FILENAME | SND_ASYNC | SND_NODEFAULT);
```

Comment:
Question: Shouldn't this be ideally, just one simple call to `PlaySoundW(L"EmptyRecycleBin", NULL, SND_ALIAS | SND_APPLICATION);` , the PlaySoundW function, thanks to the SND_APPLICATION, doing all the lifting work of opening the correct registry key based on the current name of the application?

Reply:
Note: `SND_APPLICATION` relies on `GetModuleFileName` which will break if you changed the file name from explorer.exe to something else.

Reply:
PR [#4635](https://github.com/reactos/reactos/pull/4635) is required for `SND_APPLICATION` to work too.

Reply:
I tried calling `SHEmptyRecycleBinW(NULL, NULL, SHERB_NOCONFIRMATION | SHERB_NOPROGRESSUI);` on Windows 7 from the test application named `Test_RecycleBin.exe` and created the following registry key:
![Capture_Win7](https://user-images.githubusercontent.com/86486978/206114146-9d641f3a-9b17-48a6-9c5c-b27894170559.PNG)

The function still uses the sound from Explorer so `SND_APPLICATION` flag is not used.